specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2023-07-01/virtualMachine.json

```diff
@@ -1891,6 +1891,72 @@
           "final-state-via": "location"
         }
       }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/virtualMachines/{vmName}/attachDetachDataDisks": {
+      "post": {
+        "tags": [
+          "VirtualMachines"
+        ],
+        "operationId": "VirtualMachines_AttachDetachDataDisks",
+        "description": "Attach and detach data disks to/from the virtual machine.",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "name": "vmName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the virtual machine."
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "./computeRPCommon.json#/definitions/AttachDetachDataDisksRequest"
+            },
+            "description": "Parameters supplied to the attach and detach data disks operation on the virtual machine."
+          },
+          {
+            "$ref": "../../../common-types/v1/common.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../common-types/v1/common.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "202": {
+            "description": "Accepted",
+            "schema": {
+              "$ref": "./computeRPCommon.json#/definitions/AttachDetachDataDisksResponse"
```

Comment:
>"$ref": "./computeRPCommon.json#/definitions/AttachDetachDataDisksResponse" [](http://example.com/codeflow?start=14&length=75)

202 should not have schema:
[https://github.com/Azure/azure-openapi-validator/blob/main/docs/lro-post-return.md](https://github.com/Azure/azure-openapi-validator/blob/main/docs/lro-post-return.md)

Reply:
Changes made. Currently based on team discussion with team:
Finalized HTTP Verb: HTTP POST
Response will contain both 200 and 202.
In 200, response "azure-asyncoperation" header will be provided. This is similar to start operation on the VM https://learn.microsoft.com/en-us/rest/api/compute/virtual-machines/start?tabs=HTTP#virtualmachine_start_maximumset_gen